ci/firebase_testlab.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+#
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import argparse
+import glob
+import re
+import subprocess
+import sys
+import os
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+buildroot_dir = os.path.abspath(os.path.join(script_dir, '..', '..'))
+out_dir = os.path.join(buildroot_dir, 'out')
+bucket = 'gs://flutter_firebase_testlab'
+error_re = re.compile('[EF]/flutter')
+
+
+def RunFirebaseTest(apk, results_dir):
+  try:
+    # game-loop tests are meant for OpenGL apps.
+    # This type of test will give the application a handle to a file, and
+    # we'll write the timeline JSON to that file.
+    # See https://firebase.google.com/docs/test-lab/android/game-loop
+    # Pixel 4. As of this commit, this is a highly available device in FTL.
+    subprocess.check_output([
+        'gcloud',
+        '--project', 'flutter-infra',
+        'firebase', 'test', 'android', 'run',
+        '--type', 'game-loop',
+        '--app', apk,
+        '--timeout', '2m',
+        '--results-bucket', bucket,
+        '--results-dir', results_dir,
+        '--device', 'model=flame,version=29',
+    ])
+  except subprocess.CalledProcessError as ex:
+    print(ex.output)
+    # Recipe will retry return codes from firebase that indicate an infra
+    # failure.
+    sys.exit(ex.returncode)
+
+
+def CheckLogcat(results_dir):
+  logcat = subprocess.check_output([
+      'gsutil', 'cat', '%s/%s/*/logcat' % (bucket, results_dir)
+  ])
+
+  logcat_match = error_re.match(logcat)
+  if logcat_match:
+    print('Errors in logcat:')
+    print(logcat_match)
+    sys.exit(1)
+
+
+def CheckTimeline(results_dir):
+  du = subprocess.check_output([
+      'gsutil', 'du',
+      '%s/%s/*/game_loop_results/results_scenario_0.json' % (bucket, results_dir)
+  ]).strip()
+  if du == '0':
+    print('Failed to produce a timeline.')
+    return 1
```

Comment:
Should this be a `sys.exit(1)`. The return is not checked below.

Reply:
Yeah I noticed this after uploading it and it's fixed now